@here/olp-sdk-dataservice-read/lib/utils/getTile.ts

```diff
@@ -138,13 +144,19 @@ export async function getTile(
         quadTreeIndex.subQuads.length
     ) {
         if (fetchOptions !== FetchOptions.OnlineOnly) {
-            cache.put(partitionsRequest, catalogHrn.toString(), layerId, [
-                {
-                    partition: partitionId,
-                    dataHandle: quadTreeIndex.subQuads[0].dataHandle,
-                    version: quadTreeIndex.subQuads[0].version
-                }
-            ]);
+            const subQuadsToPartitionsList = quadTreeIndex.subQuads.map(
+                item => ({
```

Comment:
What about sparse layers? If I read the code correctly, if we request a tile that doesn't exist, it fetches the index again?

Reply:
Updated the code. Returns the blob of the requested Tile or the blob of the closest parent Tile.
(or rejects with error for sure)